proto/stat.proto

```diff
@@ -46,16 +46,25 @@ message PointStat {
   // date in IOS-8601 format
   string date = 1;
   double value = 2;
+  // Full metadata of the stat
   StatMetadata metadata = 3;
+  // A hash string of the meta data, this is to be used together with a map from
+  // the hash to the full metatdata.
```

Comment:
should metadata and meta_hash be in a oneof?  Note that you can move existing fields to under a oneof ("you can safely move a single field into a new oneof " ... https://developers.google.com/protocol-buffers/docs/proto)

Reply:
Tried oneof but that makes the field nested a bit harder to access. Having the two fields here are okay too, since they can coexist potentially.

Reply:
I see. Maybe some comment on which api sets which field between metadata and meta_hash might be helpful.

Reply:
Done